### PR TITLE
Add Wayback Machine as a collector

### PR DESCRIPTION
**Pull request by jolle**
Wayback Machine by Internet Archive (http://web.archive.org/) is a "non-profit library" for archiving web pages (more: https://archive.org/about/). Even though it is mainly intended for archiving all pages under one domain, the API can be used to fetch all pages under the main domain and all subdomains. For this library (and use-case) it's easy to just loop through all pages and add the host of each URL. Aquatone is already able to not add already-added subdomains, so no need for a special system to check for duplicates.

Even though (as previously mentioned) the Wayback Machine is not the best place to find subdomains, I have seen that it has benefited in some cases. Aquatone also checks that the host is alive, so there should be no problems with subdomains that are just archived and not actually alive. I have noticed that with popular(ish) domains there are lots of archived pages and thus the loading time might be long sometimes and other times it times out, but mainly it just runs a bit slow.

### CODE REVIEW

File: lib/aquatone/collectors/wayback.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'uri'
+
+module Aquatone
+  module Collectors
+    class Crtsh < Aquatone::Collector
```
Comment (michenriksen) [inline lib/aquatone/collectors/wayback.rb]: The class name should probably be something like `WaybackMachine`

**Comment by michenriksen** [comment]
Hi @jolle, Thanks for yet another collector, awesome work! I will try it out and pull it into a release with some other stuff over the weekend.

File: lib/aquatone/collectors/wayback.rb (hunk 1 above)
Comment (jolle) [reply lib/aquatone/collectors/wayback.rb]: Oh, whoops!

**Comment by michenriksen** [comment]
Just one more nitpick: The file should be named `wayback_machine.rb` not `wayback-machine.rb` :)

Thanks!

**Comment by michenriksen** [comment]
Tried it out locally. It works, but is pretty slow and times out quite often as you said. It still finds some good subdomains though, so I will merge it in and implement a configurable timeout for the collector.

Thanks again for your contribution! 💯

**Comment by jolle** [comment]
I don't know if it would be slower or faster, but the CDX API offers out-of-the-box pagination with the `page` parameter ([more info](https://github.com/internetarchive/wayback/tree/master/wayback-cdx-server#pagination-api)). Another option would be to use the `limit` and/or `offset` parameters. Using just limit would make the response faster, but obviously, would not yield as good results as with no limit (**note**: the default limit is 150 000, so in any case, you still lose entries). Using the offset parameter though, it would be possible to loop through all entries, but then you could just use the better pagination API.

I think an option for fetching just for the first page (or _n_ (i.e. 2000) entries) or all pages (the default max of 150 000 or looping through all pages) would be good (note: an option utility _for collectors_ would be required; they only have a key utility). Using the pagination API the request flow would be `showNumPages` → loop for _n_ pages → `page=n`. I can make a PR for any of the options if any of them sound good to you!

**Pull request merged**